core/src/main/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupport.java

```diff
@@ -57,6 +60,7 @@ public static <Solution_> VariableListenerSupport<Solution_> create(InnerScoreDi
                 .toList();
         this.listVariableDescriptor = scoreDirector.getSolutionDescriptor().getListVariableDescriptor();
         this.listVariableEventList = new ArrayList<>();
+        this.unassignedValueWithEmptyInverseEntitySet = new IdentityHashSet<>();
```

Comment:
We should add a comment here, explaining why we chose this particular type.
(It's not obvious.)